src/main/java/org/folio/linked/data/controller/advice/ApiExceptionHandler.java

```diff
@@ -39,140 +24,80 @@
+    logException(e);
+    return genericBadRequestMapper.errorResponseEntity(e);
   }
 
   @ExceptionHandler(IllegalArgumentException.class)
   public ResponseEntity<ErrorResponse> handleIllegalArgumentException(IllegalArgumentException exception) {
     logException(exception);
-    return buildResponseEntity(exception, BAD_REQUEST, VALIDATION_ERROR);
-  }
-
-  @ExceptionHandler(EntityNotFoundException.class)
-  public ResponseEntity<ErrorResponse> handleEntityNotFoundException(EntityNotFoundException exception) {
-    logException(exception);
-    return buildResponseEntity(exception, NOT_FOUND, NOT_FOUND_ERROR);
-  }
-
-  @ExceptionHandler(NotFoundException.class)
-  public ResponseEntity<ErrorResponse> handleEntityNotFoundException(NotFoundException exception) {
-    log.info("Not found: {}", exception.getMessage());
-    return buildResponseEntity(exception, NOT_FOUND, NOT_FOUND_ERROR);
-  }
-
-  @ExceptionHandler(AlreadyExistsException.class)
-  public ResponseEntity<ErrorResponse> handleAlreadyExistsException(AlreadyExistsException exception) {
-    logException(exception);
-    return buildResponseEntity(exception, BAD_REQUEST, ALREADY_EXISTS_ERROR);
+    return genericBadRequestMapper.errorResponseEntity(exception);
   }
 
   @ExceptionHandler(HttpMediaTypeNotSupportedException.class)
   public ResponseEntity<ErrorResponse> handleHttpMediaTypeNotSupportedException(HttpMediaTypeNotSupportedException e) {
     logException(e);
-    return buildResponseEntity(e, BAD_REQUEST, VALIDATION_ERROR);
+    return genericBadRequestMapper.errorResponseEntity(e);
   }
 
   @ExceptionHandler(HttpMessageNotReadableException.class)
   public ResponseEntity<ErrorResponse> handlerHttpMessageNotReadableException(HttpMessageNotReadableException e) {
-    return Optional.ofNullable(e.getCause())
-      .map(Throwable::getCause)
-      .filter(IllegalArgumentException.class::isInstance)
-      .map(IllegalArgumentException.class::cast)
-      .map(this::handleIllegalArgumentException)
-      .orElseGet(() -> {
-        logException(e);
-        return buildResponseEntity(e, UNPROCESSABLE_ENTITY, VALIDATION_ERROR);
-      });
+    logException(e);
+    return genericBadRequestMapper.errorResponseEntity(e);
   }
 
   @ExceptionHandler(MissingServletRequestParameterException.class)
   public ResponseEntity<ErrorResponse> handleMissingServletRequestParameterException(
-    MissingServletRequestParameterException exception) {
-    logException(exception);
-    return buildResponseEntity(exception, BAD_REQUEST, VALIDATION_ERROR);
+    MissingServletRequestParameterException e) {
+    logException(e);
+    return genericBadRequestMapper.errorResponseEntity(e);
+  }
+
+  @ExceptionHandler(RequestProcessingException.class)
+  public ResponseEntity<ErrorResponse> handleRequestProcessingException(RequestProcessingException exception) {
+    return requestProcessingExceptionMapper.errorResponseEntity(exception);
```

Comment:
Call `logException(exception);` here also.

Reply:
I think we shouldn't do it:
RequestProcessingException is our exception, we throw it manually and we log it together with throwing.
While other exceptions are automatic, that's why we log all of them here.